### ISSUE TITLE
Lua batch_queue failed to process entries

### ISSUE THREAD

**Issue by jmicceri** [Kong/kong#10103]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

3.1.1

### Current Behavior

After upgrading to version 3.1.1 (up from 3.0.2) we starting seeing millions (~165M) of the following warning in our Kong/nginx logs:

```
3824#0: *31876612 [lua] batch_queue.lua:183: failed to process entries: nil
```

After downgrading back to `3.0.2`, the warnings went away. Are there other changes that need to be made on our end as part of the upgrade? Checking the [upgrade notes](https://docs.konghq.com/gateway/latest/upgrade/#upgrade-from-30x-to-31x), I don't see any other packages or steps that need to be taken

### Expected Behavior

Kong should be able to process all entries and there shouldn't be an increase in the number of warning logs produced 

### Steps To Reproduce

```markdown
1. Change Kong version from 3.0.2 → 3.1.1
2. Deploy Kong service (running as docker container)
3. Check NGINX logs and view increase in WARN level messages
```

**Comment by hanshuebner** [comment]
Hi @jmicceri,

can you share a couple of details of your configuration?  We're particularly interested in the plugins that you're using and the request volume.

Thank you!
Hans

**Comment by hanshuebner** [comment]
@jmicceri We've located the source of the problem and a fix is already available on the master branch (https://github.com/Kong/kong/commit/0eae6b969f6179c099b054701f453c62a3aeb088).  It will be included in the next releases.

**Comment by locao** [comment]
Closing this issue as it was fixed by #10126.
